File: sveltekit-frontend/scripts/fix-syntax-errors.mjs
#!/usr/bin/env node
/**
 * Automated Syntax Fix Script
 * Targets: Unterminated strings, missing brackets, common syntax errors
 *
 * Run: node scripts/fix-syntax-errors.mjs
 */

import fs from 'fs';
import path from 'path';
import { glob } from 'glob';

const COLORS = {
  reset: '\x1b[0m',
  green: '\x1b[32m',
  red: '\x1b[31m',
  yellow: '\x1b[33m',
  cyan: '\x1b[36m'
};

function log(message, color = 'reset') {
  console.log(`${COLORS[color]}${message}${COLORS.reset}`);
}

let stats = {
  filesScanned: 0,
  filesFixed: 0,
  unterminatedStrings: 0,
  missingBraces: 0,
  missingParens: 0,
  fixedSemicolons: 0
};

/**
 * Fix unterminated string literals
 */
function fixUnterminatedStrings(content) {
  let fixed = content;
  let fixes = 0;

  // Match unterminated strings at end of line
  const lines = content.split('\n');
  const fixedLines = lines.map((line, idx) => {
    // Check for unterminated single/double quotes
    const singleQuotes = (line.match(/'/g) || []).length;
    const doubleQuotes = (line.match(/"/g) || []).length;
    const backticks = (line.match(/`/g) || []).length;

    // Skip lines with comments
    if (line.trim().startsWith('//') || line.trim().startsWith('/*')) {
      return line;
    }

    // Fix odd number of quotes (likely unterminated)
    if (singleQuotes % 2 !== 0 && !line.includes('"') && !line.includes('`')) {
      fixes++;
      return line + "'";
    }
    if (doubleQuotes % 2 !== 0 && !line.includes("'") && !line.includes('`')) {
      fixes++;
      return line + '"';
    }
    if (backticks % 2 !== 0) {
      fixes++;
      return line + '`';
    }

    return line;
  });

  stats.unterminatedStrings += fixes;
  return fixedLines.join('\n');
}

/**
 * Fix missing closing braces
 */
function fixMissingBraces(content) {
  const openBraces = (content.match(/{/g) || []).length;
  const closeBraces = (content.match(/}/g) || []).length;

  if (openBraces > closeBraces) {
    const diff = openBraces - closeBraces;
    stats.missingBraces += diff;
    return content + '\n' + '}'.repeat(diff) + '\n';
  }

  return content;
}

/**
 * Fix missing closing parentheses
 */
function fixMissingParens(content) {
  const openParens = (content.match(/\(/g) || []).length;
  const closeParens = (content.match(/\)/g) || []).length;

  if (openParens > closeParens) {
    const diff = openParens - closeParens;
    stats.missingParens += diff;
    return content + ')'.repeat(diff);
  }

  return content;
}

/**
 * Fix missing semicolons (common pattern)
 */
function fixMissingSemicolons(content) {
  let fixed = content;
  let fixes = 0;

  // Add semicolon after variable declarations without it
  fixed = fixed.replace(/(const|let|var)\s+\w+\s*=\s*[^;]+$/gm, (match) => {
    if (!match.trim().endsWith(';') && !match.trim().endsWith(',')) {
      fixes++;
      return match + ';';
    }
    return match;
  });

  stats.fixedSemicolons += fixes;
  return fixed;
}

/**
 * Main fix function
 */
async function fixFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    const originalContent = content;

    // Apply fixes in order
    content = fixUnterminatedStrings(content);
    content = fixMissingBraces(content);
    content = fixMissingParens(content);
    content = fixMissingSemicolons(content);

    // Only write if changes were made
    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8');
      stats.filesFixed++;
      return true;
    }

    return false;
  } catch (error) {
    log(`Error processing ${filePath}: ${error.message}`, 'red');
    return false;
  }
}

/**
 * Main execution
 */
async function main() {
  log('\n╔══════════════════════════════════════════════════╗', 'cyan');
  log('║  Automated Syntax Error Fix - YoRHa Detective  ║', 'cyan');
  log('╚══════════════════════════════════════════════════╝\n', 'cyan');

  // Find all TypeScript files
  log('🔍 Scanning for TypeScript files...', 'cyan');
  const files = await glob('src/**/*.ts', {
    ignore: [
      '**/node_modules/**',
      '**/.svelte-kit/**',
      '**/build/**',
      '**/*.d.ts' // Skip type definitions
    ]
  });

  stats.filesScanned = files.length;
  log(`📁 Found ${files.length} TypeScript files\n`, 'green');

  // Process files
  log('🔧 Applying automated fixes...\n', 'yellow');

  for (const file of files) {
    const fixed = await fixFile(file);
    if (fixed) {
      const relPath = path.relative(process.cwd(), file);
      log(`  ✅ Fixed: ${relPath}`, 'green');
    }

    // Progress indicator
    if (stats.filesScanned > 0 && stats.filesFixed % 10 === 0) {
      const progress = ((stats.filesFixed / stats.filesScanned) * 100).toFixed(1);
      process.stdout.write(`\r  Progress: ${progress}%`);
    }
  }

  console.log('\n');

  // Report
  log('═══════════════════════════════════════', 'cyan');
  log('REPAIR SUMMARY', 'cyan');
  log('═══════════════════════════════════════', 'cyan');
  log(`Files Scanned:         ${stats.filesScanned}`, 'blue');
  log(`Files Fixed:           ${stats.filesFixed}`, 'green');
  log(`Unterminated Strings:  ${stats.unterminatedStrings}`, 'yellow');
  log(`Missing Braces:        ${stats.missingBraces}`, 'yellow');
  log(`Missing Parentheses:   ${stats.missingParens}`, 'yellow');
  log(`Fixed Semicolons:      ${stats.fixedSemicolons}`, 'yellow');
  log('═══════════════════════════════════════\n', 'cyan');

  if (stats.filesFixed > 0) {
    log('✅ Automated fixes applied successfully!', 'green');
    log('\n💡 Next steps:', 'cyan');
    log('   1. Run: npm run check:typescript', 'blue');
    log('   2. Review changes: git diff', 'blue');
    log('   3. Commit if satisfied: git add -A && git commit -m "fix: automated syntax repairs"', 'blue');
  } else {
    log('ℹ️  No fixable issues found (or all already fixed)', 'yellow');
  }

  console.log();
}

// Run
main().catch(error => {
  log(`\n❌ Script failed: ${error.message}`, 'red');
  console.error(error);
  process.exit(1);
});

File: sveltekit-frontend/docs_readme/deeds_labs_archive/fix-syntax-errors.py
#!/usr/bin/env python3
import os
import re
import sys

def fix_syntax_errors(filepath):
    """Fix common TypeScript syntax errors"""
    with open(filepath, 'r', encoding='utf-8') as f:
        lines = f.readlines()

    original = lines.copy()
    modified = False

    for i, line in enumerate(lines):
        # Fix unclosed map/filter/reduce parentheses
        # Pattern: .map(...) => ({...}) should be .map(...) => ({...}))
        if re.search(r'\.map\([^)]+\) => \([^)]+\}\);?$', line):
            if not line.rstrip().endswith('));') and not line.rstrip().endswith('));'):
                lines[i] = line.rstrip()[:-1] + '));\n'
                modified = True

        # Fix Promise<Type<SubType> should be Promise<Type<SubType>>
        if 'Promise<' in line and line.count('<') > line.count('>'):
            # Count angle brackets
            open_brackets = line.count('<')
            close_brackets = line.count('>')
            if open_brackets == close_brackets + 1:
                # Find position to insert missing >
                if ' {' in line:
                    lines[i] = line.replace(' {', '> {', 1)
                    modified = True
                elif ');' in line:
                    lines[i] = line.replace(');', '>);', 1)
                    modified = True

        # Fix async function return types
        if re.search(r':\s*Promise<[^>]+<[^>]+>\s*{', line):
            lines[i] = re.sub(r'(:\s*Promise<[^>]+<[^>]+)>\s*{', r'\1>> {', line)
            modified = True

    if modified:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.writelines(lines)
        return True
    return False

def find_unclosed_brackets(filepath):
    """Find files with unbalanced brackets"""
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Skip comments
    content = re.sub(r'//.*?\n', '\n', content)
    content = re.sub(r'/\*.*?\*/', '', content, flags=re.DOTALL)

    # Count brackets
    open_parens = content.count('(')
    close_parens = content.count(')')
    open_braces = content.count('{')
    close_braces = content.count('}')
    open_brackets = content.count('[')
    close_brackets = content.count(']')

    issues = []
    if open_parens != close_parens:
        issues.append(f"Parentheses: {open_parens} open, {close_parens} close")
    if open_braces != close_braces:
        issues.append(f"Braces: {open_braces} open, {close_braces} close")
    if open_brackets != close_brackets:
        issues.append(f"Brackets: {open_brackets} open, {close_brackets} close")

    return issues

def main():
    fixed_count = 0
    total_files = 0
    problematic_files = []

    for root, dirs, files in os.walk('src/lib'):
        for file in files:
            if file.endswith('.ts'):
                filepath = os.path.join(root, file)
                total_files += 1

                # First check for issues
                issues = find_unclosed_brackets(filepath)
                if issues:
                    problematic_files.append((filepath, issues))

                # Try to fix
                if fix_syntax_errors(filepath):
                    fixed_count += 1
                    print(f"Fixed: {filepath}")

    print(f"\nProcessed {total_files} files, fixed {fixed_count} files")

    if problematic_files:
        print(f"\nFiles with unbalanced brackets ({len(problematic_files)}):")
        for filepath, issues in problematic_files[:10]:
            print(f"  {filepath}: {', '.join(issues)}")

if __name__ == "__main__":
    main()